### PR TITLE
Register (for Coqlib.ref_lib) several base datatypes of stdlib

### PR DESCRIPTION
**Pull request by lthms**
Following a discussion I had with @ejgallego, I try to add `Register` call for base dataypes of the stdlib.

The second commit is potentially a breaking change, since to remain consistent I changed the path/namespace for `string` constructors. As far as I can tell, it is not yet used in the Coq codebase anywhere, but some external plugins may have start relying on the old value?

Let me know if this kind of change is OKay, and if you want me to have a deeper look into stdlib codebase.

---------------

**Kind:** bug fix

- [ ] Corresponding documentation was added / updated (including any warning and error messages added / removed / modified).
- [ ] Entry added in the changelog (see https://github.com/coq/coq/tree/master/doc/changelog#unreleased-changelog for details).

### CODE REVIEW

**Comment by lthms** [comment]
Dear Coq developers,

I am sorry I let this PR slowly dies here.

Since I propose this change, my opinion on this matter has slightly evolves and before going any further I am curious to gather feedback.

I once thought it was desirable for `Register` to be used only once for each constant; I am less sure now. I guess it’s okay-ish for a plugin which needs to use `Coqlib.ref_lib` to register a name to its convenience if needed, even if it is for the standard library.

What do you think?

**Comment by ejgallego** [comment]
Hi @lthms , I dunno; I guess both approaches are valid; tho I think that for generic datatypes as the ones here it makes sense to register them at the core.

**Comment by lthms** [comment]
Makes sense too. What should we do? Merge the PR as-is? I can try to have a more systematic approach before merging, though.

**Comment by maximedenes** [comment]
> I think that for generic datatypes as the ones here it makes sense to register them at the core.

Yes, I also think that's the general guideline.

This PR looks good enough to ask for a review IMHO. An incremental approach seems fine in this case.

**Comment by Zimmi48** [comment]
For the record this PR is consistent with what was recently done for the real library (#11746).

**Comment by lthms** [comment]
> This PR looks good enough to ask for a review IMHO.

If it is good enough for you, it is good enough for me! :)

**Comment by Zimmi48** [comment]
Maybe you should backtrack on the renamings. It seems to have broken the test-suite (test `output/ExtractionString.v`, introduced in #10486). I haven't looked into it closer than this.

**Comment by lthms** [comment]
That makes sense due to the particular semantics of `core` I guess. I will try to investigate, but as you suggested the simple way to go is probably to just remove the second commit.

**Review by JasonGross — CHANGES_REQUESTED** [review]
It's not clear to me that ascii and string should live in `plugins.syntax`.  I think you should leave them in `core`, but I'm open to arguments to the contrary.  In any case, if you want to rename the constructors (it's not entirely clear to me that you should), then please add a changelog entry and add back the old registration in the relevant Compat file, and update the OCaml code which makes use of these constants.

**Comment by lthms** [comment]
Thanks for your feedback, @JasonGross! I understand your position, and even if I had my logical reason to propose this change (mostly, for consistency wrt. what I had seen elsewhere in the Coq codebase), I don’t think anymore this is a big deal that everything is not consistent. The way I see it, there is no way for a plugin developer to know if a given name has been registered before (except for reading the source), and there is no guarantee a name valid for a given version will remains to another, so I believe the best practice would be for developers to register what they need with the value they desire.

So basically, I propose to limit this PR to the first commit, as suggested by @Zimmi48

**Comment by herbelin** [comment]
As a @coq/stdlib-maintainers, I may serve as assignee. I must confess however that I did not follow the `Register` thing and corresponding naming convention.

I'm sympathesizing with @JasonGross' arguments about `string`, so the PR looks good to me with only its first commit. If @jasongross gives me the green light that his request has been answered and that he's ok with the PR, I would merge in a couple of days. Other participants to the discussion, don't hesitate to comment more if you wish.

**Review by JasonGross — APPROVED** [review]

**Comment by lthms** [comment]
Thanks @herbelin 

> I must confess however that I did not follow the `Register` thing and corresponding naming convention.

I don’t thing there is any well-established convention for now. Maybe we could use this occasion to update the manual to fix this? or it may be delayed in another PR, maybe.

**Comment by Zimmi48** [comment]
Let's not overload this PR which seems otherwise ready to be merged.

**Comment by herbelin** [comment]
Failure of CoRN is presumably unrelated. Will merge later in the day.

**Pull request merged**
